### PR TITLE
Add equivalent ripped groove

### PR DESCRIPTION
**Pull request by RichardPfr**

### CODE REVIEW

File: pyroll/core/grooves/equivalent_ripped_groove.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        # height of the equivalent circle segment on top of the base body
+        equivalent_circle_segment_height = ribbed_circle_segment_height * width_distance_ratio
+
+
+        # diagonal width of the base body
+        base_body_diagonal_width = base_body_height * np.sqrt(2)
+
+        # help triangle - interim calculation for later use
+        help_triangle_inner_angle = 180 - (45 + (180 - np.rad2deg(np.arcsin(((base_body_diagonal_width / 2) *
```
Comment (axtimhaus) [inline pyroll/core/grooves/equivalent_ripped_groove.py:70]: please avoid forth and back conversion of angles, reformulate in radians

File: tests/grooves/test_equivalent_ribbed_groove.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def check(g):
+    plt.figure(dpi=300)
+    plt.axes().set_aspect("equal")
+    plt.plot(*g.contour_points.T)
+    plt.show()
+
+    assert isclose(g.r1, 0.2)
+    assert isclose(g.r3, 3.45)
+    assert isclose(g.usable_width, 13.6788)
+    assert isclose(g.depth, 5.5091)
```
Comment (axtimhaus) [inline tests/grooves/test_equivalent_ribbed_groove.py:17]: These checks are just for the given input parameters, add more checks for calculated measures

**Review by axtimhaus — CHANGES_REQUESTED** [review]

**Review by ChRen95 — CHANGES_REQUESTED** [review]

File: pyroll/core/grooves/equivalent_ripped_groove.py [hunk 3]
```diff
@@ -47,34 +47,17 @@ def __init__(
...
-                                    np.sin(np.deg2rad(45))) / nominal_outer_radius))))
-
-        # base width of the new circle segment on top of the base body
+        help_triangle_inner_angle = np.pi - (np.pi / 4 + (np.pi - np.arcsin(((base_body_diagonal_width / 2) *
+                                    np.sin(np.deg2rad(45))) / nominal_outer_radius)))
         circle_segment_base_width = base_body_height - (2 * ((nominal_outer_radius *
-                                    np.sin(np.deg2rad(help_triangle_inner_angle))) / np.sin(np.deg2rad(45))))
-
-        # r2 calculated from the base width and height of the new circle segment
+                                    np.sin(help_triangle_inner_angle)) / np.sin(np.deg2rad(45))))
```
Comment (ChRen95) [inline pyroll/core/grooves/equivalent_ripped_groove.py:60]: Warum hier deg2rad(45)  und nicht np.pi / 2?

**Review by ChRen95 — CHANGES_REQUESTED** [review]
See comments

File: pyroll/core/grooves/equivalent_ripped_groove.py (hunk 3 above)
Comment (RichardPfr) [reply pyroll/core/grooves/equivalent_ripped_groove.py:60]: I used it for better understandability on my side, but changed it to np.pi / 4 now.

**Review by axtimhaus — APPROVED** [review]

**Review by ChRen95 — APPROVED** [review]

**Pull request merged**